src/rng.cpp

```diff
@@ -7,40 +7,70 @@
 #include <cstdlib>
 #include <random>
 
-long rng( long val1, long val2 )
+static std::mt19937 &rng_get_engine();
+static std::uniform_int_distribution<long> rng_int_dist;
+static std::uniform_real_distribution<double> rng_real_dist;
+static std::normal_distribution<double> rng_normal_dist;
+
+uintmax_t rng_bits()
```

Comment:
Rather than rolling your own implementation here you could use `std::uniform_int_distribution<uintmax_t>(0, std::numeric_limits<uintmax_t>::max())`.  It may be slower, but probably not significantly so.